dev/bots/test.dart

```diff
@@ -744,6 +747,109 @@ Future<void> _runWebUnitTests() async {
   await selectSubshard(subshards);
 }
 
+/// Coarse-grained integration tests running on the Web.
+///
+/// These tests are sharded into [kWebLongRunningTestShardCount] shards.
+Future<void> _runWebLongRunningTests() async {
+  final List<ShardRunner> tests = <ShardRunner>[
+    () => _runGalleryE2eWebTest('debug'),
```

Comment:
It will be one shard, but it will be split into `kWebLongRunningTestShardCount` subshards, which we can tweak over time. The setup we use for the `web_tests` shard is ideal for this use-case. `web_integration_tests` doesn't have subshards, so it won't scale.

Reply:
web_tests are setup to shards from config file as well. I might be missing something but I didn't get the usage for `kWebLongRunningTestShardCount`. Also the comment might also be misleading since the number we set to the config will determine the number of subshards not the constant we add here.

```
/// These tests are sharded into [kWebLongRunningTestShardCount] shards.
```

Note for later: I added another recipe to the engine side for running web_tests. we might still need to make some modifications on that.

Reply:
Check out the line below: `await _selectIndexedSubshard(tests, kWebLongRunningTestShardCount);`. `_selectIndexedSubshard` uses the count to divide the total number of tests (6) into subshards (3) then run one of them (2 tests per subshard). The hard-coded count is needed because `test.dart` doesn't receive it from LUCI configs (i.e. the number 3). Started a discussion on #hackers-infra.

Reply:
I see thanks for the explanation. For using the same value with LUCI infra we have 2 options:

-> We can use the value from flutter/infra as source of truth. We modify the recipes to write an env variable. This code reads that env variable.
-> We can write it to a yaml file in this repo (so the source of truth is flutter/flutter). We can read the yaml file from the LUCI recipe (we need to be careful with this design since flutter.py and flutter_drone.py is very generic, we don't want to add a random if/else there).

Both options can easily be tested with `led`.

@godofredoc @christopherfujino which do you think would be the best approach.

Reply:
I'd vote for adding the value to flutter/infra and use it in the script. As part of the LUCI transition we would like to keep everything related to infra resources allocation/usage in the LUCI space to have more visibility and better management of the resources.

Reply:
This sounds like something outside the scope of my PR. However, I'd vote for adding these to flutter/flutter. Currently flutter/infra leads to non-hermetic builds (you build a PR with 5 subshards and see subshard 3 fail; you build it again but someone changed the shard count to 8 in flutter/infra, and now subshards 1 and 6 fail; you pull your hair).

If the system is designed correctly, adding new shards or subshards should not affect infra resource allocation. Instead a PR should get up to a limited amount of resources (the limit can change depending on the overall system load), and schedule the shards/subshards (no matter how many of them) on top of the available resources, similar to how an OS will schedule processes/threads on a limited number of CPU cores, no matter how many processes/threads there are, and similar to how Bazel schedules thousands of build targets with concurrency limited to your CPU core count.

However, if the infra is unable to implement per-PR scheduling yet, we could require that the file that controls sharding is reviewed by someone from the infra team.